torchreid/engine/engine.py

```diff
@@ -223,22 +225,20 @@ def run(
             raise ValueError('visrank can be set to True only if test_only=True')
 
         if test_only:
-            results = self.test(
-                                0,
-                                dist_metric=dist_metric,
-                                normalize_feature=normalize_feature,
-                                visrank=visrank,
-                                visrank_topk=visrank_topk,
-                                save_dir=save_dir,
-                                use_metric_cuhk03=use_metric_cuhk03,
-                                ranks=ranks,
-                                rerank=rerank,
-                            )
-            return results
-
-        if self.writer is None:
-            log_dir = tb_log_dir if len(tb_log_dir) else save_dir
-            self.writer = SummaryWriter(log_dir=log_dir)
+            test_results = self.test(
+                                     0,
+                                     dist_metric=dist_metric,
+                                     normalize_feature=normalize_feature,
+                                     visrank=visrank,
+                                     visrank_topk=visrank_topk,
+                                     save_dir=save_dir,
+                                     use_metric_cuhk03=use_metric_cuhk03,
+                                     ranks=ranks,
+                                     rerank=rerank,
+                                    )
+            return test_results
+
+        self.writer = tb_writer
```

Comment:
Where the tensorboard writer is being built now? As I can see, engine.run() goes w/o writer

Reply:
Indeed, forgot about that. Writer was added to main